.github/workflows/ci-build.yaml

```diff
@@ -462,51 +446,31 @@ jobs:
         run: |
           git config --global user.name "John Doe"
           git config --global user.email "john.doe@example.com"
-      - name: Pull Docker image required for tests
-        run: |
-          docker pull ghcr.io/dexidp/dex:v2.40.0
-          docker pull argoproj/argo-cd-ci-builder:v1.0.0
-          docker pull redis:7.0.15-alpine
       - name: Create target directory for binaries in the build-process
         run: |
           mkdir -p dist
           chown runner dist
-      - name: Run E2E server and wait for it being available
-        timeout-minutes: 30
-        run: |
-          set -x
-          # Something is weird in GH runners -- there's a phantom listener for
-          # port 8080 which is not visible in netstat -tulpen, but still there
-          # with a HTTP listener. We have API server listening on port 8088
-          # instead.
-          make start-e2e-local COVERAGE_ENABLED=true 2>&1 | sed -r "s/[[:cntrl:]]\[[0-9]{1,3}m//g" > /tmp/e2e-server.log &
-          count=1
-          until curl -f http://127.0.0.1:8088/healthz; do
-            sleep 10;
-            if test $count -ge 180; then
-              echo "Timeout"
-              exit 1
-            fi
-            count=$((count+1))
-          done
+      - name: Install self-signed certs
+        run: |
+          curl -JLO "https://dl.filippo.io/mkcert/latest?for=linux/amd64"
+          chmod +x mkcert-v*-linux-amd64
+          sudo cp mkcert-v*-linux-amd64 /usr/local/bin/mkcert
+          mkcert -cert-file test/fixture/certs/argocd-e2e-server.crt.pem -key-file test/fixture/certs/argocd-e2e-server.key.pem argocd-e2e-server localhost 127.0.0.1 ::1
+          mkcert -cert-file test/fixture/certs/argocd-test-client.crt.pem -key-file test/fixture/certs/argocd-test-client.key.pem -client argocd-e2e-server localhost 127.0.0.1 ::1          
+          mkcert -install
+          cp "$(mkcert -CAROOT)"/rootCA.pem test/fixture/certs/argocd-test-ca.pem
+          cp "$(mkcert -CAROOT)"/rootCA-key.pem test/fixture/certs/argocd-test-ca.key.pem
       - name: Run E2E testsuite
         run: |
           set -x
           make test-e2e-local
-          goreman run stop-all || echo "goreman trouble"
```

Comment:
This was added to make sure the argocd processes would shutdown gracefully and have the golang coverage from the e2e "flushed" to the files. We should just confirm that test-containers sends the proper signals to cause the flush to happen.